### PR TITLE
Basic stack pooling allocator for Asyncify stacks

### PR DESCRIPTION
**Pull request by dhil**
This patch adds a basic stack pooling allocator for Asyncify stacks. It is togglable by setting the build option `USE_STACK_POOLING=1`. The maximum number of stacks is controllable via the option `STACK_POOL_SIZE`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ASYNCIFY_DEFAULT_STACK_SIZE?=2097152
+USE_STACK_POOLING?=0
```
Comment (frank-emrich) [inline Makefile:2]: Is this supposed to eventually control stack pooling for both asyncify and wasmfx simultaneously? If not, both of these should be prefixed with `ASYNCIFY_`

File: src/asyncify/asyncify_impl.c [hunk 2]
```diff
@@ -86,13 +86,42 @@ static void fiber_stack_free(struct fiber_stack fiber_stack) {
   free(fiber_stack.buffer);
 }
 
+#if defined USE_STACK_POOLING && USE_STACK_POOLING == 1
```
Comment (frank-emrich) [inline src/asyncify/asyncify_impl.c:89]: It seems to be more idiomatic to use `#ifdef` for feature flags in C.
You could do the following in the makefile, similar to what I'm doing for the shadow stack:
```
ifeq ($(USE_STACK_POOLING),1)
  STACK_POOLING_FLAGS=-DUSE_STACK_POOLING -DSTACK_POOL_SIZE=$(STACK_POOL_SIZE)
else
  STACK_POOLING_FLAGS=
endif
```

**Review by frank-emrich — APPROVED** [review]
LGTM modulo nits

File: Makefile (hunk 1 above)
Comment (dhil) [reply Makefile:2]: I'd like to control both implementations, eventually. For that to happen, we need to expose the WasmFX config in the C API of Wasmtime.

File: src/asyncify/asyncify_impl.c (hunk 2 above)
Comment (dhil) [reply src/asyncify/asyncify_impl.c:89]: On second thought, we could fuse this into a single flag: `USE_STACK_POOLING=0` means on the demand allocation, and `USE_STACK_POOLING=N` for `N > 0` means stack pooling with max `N` stacks for now?

Comment (frank-emrich) [reply src/asyncify/asyncify_impl.c:89]: I resisted that temptation when it came to the shadow stack flags... Having a boolean-sounding flag whose non-zero values are not equivalent seems unintuitive.

Comment (frank-emrich) [reply src/asyncify/asyncify_impl.c:89]: Actually, how about using `STACK_POOL_SIZE` as the name of the unified flag? Having `0` mean "disabled" then sounds reasonable

Comment (dhil) [reply src/asyncify/asyncify_impl.c:89]: We can do that!

**Pull request merged**